azure-mgmt-reservations/azure/mgmt/reservations/azure_reservation_api.py

```diff
@@ -0,0 +1,265 @@
+from msrestazure.azure_operation import AzureOperationPoller
+import uuid
+from .operations.reservation_order_operations import ReservationOrderOperations
+from .operations.reservation_operations import ReservationOperations
+from .operations.operation_operations import OperationOperations
+from . import models
+
+
+class AzureReservationAPIConfiguration(AzureConfiguration):
+    """Configuration for AzureReservationAPI
+    Note that all parameters used to create this instance are saved as instance
+    attributes.
+
+    :param credentials: Credentials needed for the client to connect to Azure.
+    :type credentials: :mod:`A msrestazure Credentials
+     object<msrestazure.azure_active_directory>`
+    :param str base_url: Service URL
+    """
+
+    def __init__(
+            self, credentials, base_url=None):
+
+        if credentials is None:
+            raise ValueError("Parameter 'credentials' must not be None.")
+        if not base_url:
+            base_url = 'https://management.azure.com'
+
+        super(AzureReservationAPIConfiguration, self).__init__(base_url)
+
+        self.add_user_agent('azurereservationapi/{}'.format(VERSION))
+        self.add_user_agent('Azure-SDK-For-Python')
+
+        self.credentials = credentials
+
+
+class AzureReservationAPI(object):
+    """This API describe Azure Reservation
+
+    :ivar config: Configuration for client.
+    :vartype config: AzureReservationAPIConfiguration
+
+    :ivar reservation_order: ReservationOrder operations
+    :vartype reservation_order: reservations.operations.ReservationOrderOperations
+    :ivar reservation: Reservation operations
+    :vartype reservation: reservations.operations.ReservationOperations
+    :ivar operation: Operation operations
+    :vartype operation: reservations.operations.OperationOperations
+
+    :param credentials: Credentials needed for the client to connect to Azure.
+    :type credentials: :mod:`A msrestazure Credentials
+     object<msrestazure.azure_active_directory>`
+    :param str base_url: Service URL
+    """
+
+    def __init__(
+            self, credentials, base_url=None):
+
+        self.config = AzureReservationAPIConfiguration(credentials, base_url)
+        self._client = ServiceClient(self.config.credentials, self.config)
+
+        client_models = {k: v for k, v in models.__dict__.items() if isinstance(v, type)}
+        self.api_version = '2017-11-01'
+        self._serialize = Serializer(client_models)
+        self._deserialize = Deserializer(client_models)
+
+        self.reservation_order = ReservationOrderOperations(
+            self._client, self.config, self._serialize, self._deserialize)
+        self.reservation = ReservationOperations(
+            self._client, self.config, self._serialize, self._deserialize)
+        self.operation = OperationOperations(
+            self._client, self.config, self._serialize, self._deserialize)
+
+    def register_subscription(
```

Comment:
This operation is already covered by the azure-mgmt-resource package:
https://docs.microsoft.com/en-us/python/api/azure.mgmt.resource.resources.v2017_05_10.operations.providersoperations?view=azure-python#azure_mgmt_resource_resources_v2017_05_10_operations_ProvidersOperations_register

Moreover, Autorest runtime handles auto-registration automatically:
https://github.com/Azure/msrestazure-for-python#2017-06-19-version-049

Please remove this operation from your Swagger and from here.